cupy/core/elementwise.pxi

```diff
@@ -106,9 +111,14 @@ cdef _int_type = _int_iinfo.dtype.type
 
 cpdef _python_scalar_to_numpy_scalar(x):
     # Note that isinstance(x, six_integer_types) matches with bool.
-    if isinstance(x, bool):
+    typ = type(x)
```

Comment:
Is it safe? `x` could be of some type derived from Python built-in scalars.

Reply:
Previous code uses `_python_type_to_numpy_type`.

Reply:
Do you mean we are not going to support such types?

Reply:
OK, I will add fallback code.

Reply:
(By the way, I'm not insisting we should support them. But if we are not going to support them, it's better to make consensus and write in the documentation.)

Reply:
I found [this condition](https://github.com/cupy/cupy/pull/1318/files#diff-0dcdc9e12dc6104b40e22595b3566bafR317).
Already, the input data is filtered by it.

Reply:
I can't follow the link you pasted, but if there's assumption on this function, it's better to write in a comment.

Reply:
Sorry, Please check this link.
https://github.com/okuta/cupy/blob/0a87a602e4b3a7276fd92bf753e94fcaa86288f2/cupy/core/elementwise.pxi#L152